mlir/lib/Dialect/LLVMIR/IR/LLVMTypes.cpp

```diff
@@ -232,7 +232,10 @@ LLVMFunctionType::getChecked(function_ref<InFlightDiagnostic()> emitError,
 
 LLVMFunctionType LLVMFunctionType::clone(TypeRange inputs,
                                          TypeRange results) const {
-  assert(results.size() == 1 && "expected a single result type");
+  if (results.size() != 1 || !isValidResultType(results[0]))
+    return {};
+  if (!llvm::all_of(inputs, isValidArgumentType))
+    return {};
   return get(results[0], llvm::to_vector(inputs), isVarArg());
```

Comment:
Can't we just use `getChecked` here?

Reply:
It emits a diagnostic that we don't want here. And we don't have an equivalent of `llvm::nulls` for diagnostics were we could just send it to the void. Overriding the handler locally sounds a bit too heavy here...